core/node/proof_data_handler/src/tee_request_processor.rs

```diff
@@ -41,32 +38,77 @@ impl TeeRequestProcessor {
     ) -> Result<Json<TeeProofGenerationDataResponse>, RequestProcessorError> {
         tracing::info!("Received request for proof generation data: {:?}", request);
 
-        let mut connection = self
-            .pool
-            .connection()
-            .await
-            .map_err(RequestProcessorError::Dal)?;
-
-        let l1_batch_number_result = connection
-            .tee_proof_generation_dal()
-            .get_next_batch_to_be_proven(request.tee_type, self.config.proof_generation_timeout())
-            .await
-            .map_err(RequestProcessorError::Dal)?;
-
-        let l1_batch_number = match l1_batch_number_result {
-            Some(number) => number,
-            None => return Ok(Json(TeeProofGenerationDataResponse(None))),
+        let mut min_batch_number: Option<L1BatchNumber> = None;
+        let mut missing_range: Option<(L1BatchNumber, L1BatchNumber)> = None;
```

Comment:
Do you think the additional logic to handle the range scenario is worth it? I'd just log a message if the blob for some batch number is missing w/o trying to keep track of ranges. The only reason I can think of why there are gaps is when we have a situation like now, where we deploy part of the code, accumulate lots of "to_be_proven" batches and only later activate the logic to actually produce a proof. In normal operation, there shouldn't really be any gap and/or missing blob. Not sure.

Reply:
> (...) The only reason I can think of why there are gaps is when we have a situation like now, where we deploy part of the code, accumulate lots of "to_be_proven" batches and only later activate the logic to actually produce a proof. (...)

That reason alone is enough for me to want this logic implemented. I can also imagine a scenario where `proof_data_handler` module temporarily loses access to GCP for whatever reason (like authorization issues, GCP being down, or something else). In that case, we’d get flooded with logs.

Anyway, I’m not strongly attached to this. It’s just an implementation detail that I can easily roll back with [`git revert 35cb71c86781ac8aa3d4d5c7a2ea096b30befaa9`](https://github.com/matter-labs/zksync-era/pull/2674/commits/35cb71c86781ac8aa3d4d5c7a2ea096b30befaa9).

Any strong opinions on this, @slowli? We need a tie-breaker here. I’ll go with whatever suggestion you have, just to get it merged today. :P

Reply:
IMO, it's reasonable to avoid log spamming if it doesn't introduce much complexity.